src/check_executor.rs

```diff
@@ -91,15 +108,18 @@ pub fn queue_check(
         config,
         resolve_tx,
     };
-    sender
+    check_sender
+        .sender
         .send(scheduled_check)
         .expect("Failed to queue ScheduledCheck");
+    check_sender.queue_size.fetch_add(1, Ordering::SeqCst);
```

Comment:
Using sequential consistency for all the memory ordering operations here based on the answers in this question https://stackoverflow.com/questions/9553591/c-stdatomic-what-is-stdmemory-order-and-how-to-use-them